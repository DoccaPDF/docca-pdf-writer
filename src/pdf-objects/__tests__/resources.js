import Resources from '../resources';

const resources = Resources({
  id: 2,
  ProcSet: ['PDF', 'Text', 'ImageB', 'ImageC', 'ImageI'],
  Font: { F1: { id: 3 } },
  XObject: { X1: { id: 5 } }
});

describe('pdf-objects', () => {
  describe('resources', () => {
    it('creates a new object', () => {
      expect(resources).toMatchSnapshot();
    });

    it('flattens keys values', () => {
      expect(resources.flattenKeys()).toMatchSnapshot();
    });

    it('returns a PDF object', () => {
      expect(resources.toPDF()).toMatchSnapshot();
    });
  });
});
