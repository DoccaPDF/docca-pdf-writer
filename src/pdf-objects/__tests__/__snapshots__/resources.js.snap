// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`pdf-objects resources creates a new object 1`] = `
Object {
  "Font": Object {
    "F1": "3 0 R",
  },
  "ProcSet": Array [
    "PDF",
    "Text",
    "ImageB",
    "ImageC",
    "ImageI",
  ],
  "XObject": Object {
    "X1": "5 0 R",
  },
  "id": 2,
}
`;

exports[`pdf-objects resources flattens keys values 1`] = `
Object {
  "Font": "<<
/F1 3 0 R
>>
",
  "ProcSet": "[ /PDF /Text /ImageB /ImageC /ImageI ]",
  "XObject": "<<
/X1 5 0 R
>>
",
}
`;

exports[`pdf-objects resources returns a PDF object 1`] = `
"2 0 obj
<<
/ProcSet [ /PDF /Text /ImageB /ImageC /ImageI ]
/Font <<
/F1 3 0 R
>>

/XObject <<
/X1 5 0 R
>>

>>
endobj
"
`;
